### PR TITLE
The Resource Wars: Adds two new events to aid the "lowpop: no miner = no resource" issue

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

This PR adds two new events to give the station a passive source of resources even if they don't have any miners on deck due to low-pop or dead shaft miners.

These events have tons of randomized flavor in their announcements to avoid repetition and add to the world-building. It's fun!

### Resource Pods
2 to 5 drop pods of resource crates will be launched in a random room on the station. These pods can be syndicate, nanotrasen, or special! These pods can contain a random amount of basic basic resources (metal, glass, gold, silver, etc), rare resources, or special goods depending on the owner of the pod.

Nanotrasen pods can contain bonus advanced building materials - plasteel, plasma-glass.
Syndicate pods can contain plastitanium and plastinium glass. They also might contain a few syndicate contraband items...
Special pods can contain some miscellaneous rare sheets of resources like alien alloy, or maybe they'll be extra special and contain things related to the source of the pod.

### Resource Drift
3 to 8 crates will be thrown towards the station from space. These crates will just be normal resources like metal, glass, gold, diamonds, etc. Most of them will hit the side of the station, some will fly off course, but no matter what it's up to the crew to EVA to retrieve the crates.

Along with the crates, some amount of space dust will be catapulted towards the station, to keep engineers on their toes. Don't be outside while the crates are being launched, it's not saaafe!

### CODE REVIEW

File: jollystation_modules/code/modules/events/resource_drift.dm [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+		return MAP_ERROR
+	if(!picked_crates.len)
+		CRASH("Resource drift: No list of picked crates found.")
+
+	// Spawn some debris in based on the number of caches we're throwing.
+	switch(amt_debris)
+		if(MAJOR_DEBRIS)
+			spawn_meteors(num_caches * 3, GLOB.meteorsC)
+		if(MINOR_DEBRIS)
+			spawn_meteors(num_caches * 2, GLOB.meteorsC)
```
Comment (MrMelbert) [inline jollystation_modules/code/modules/events/resource_drift.dm:66]: This event can spawn quite a lot of space dust, let me know if it's excessive and should be toned down

File: jollystation_modules/code/modules/events/resource_pods.dm [hunk 2]
```diff
@@ -0,0 +1,193 @@
+/// This event spawns multiple cargo pods containing a few resources.
+/datum/round_event_control/resource_pods
+	name = "Resource Pods"
+	typepath = /datum/round_event/resource_pods
+	// Relatively common, no maximum amount, can start early
+	weight = 25
+	earliest_start = 5 MINUTES
```
Comment (MrMelbert) [inline jollystation_modules/code/modules/events/resource_pods.dm:7]: This event has very high weight and can happen very early with no (reachable) max, let me know if it's too excessive and needs to be cut back

File: jollystation_modules/code/modules/events/resource_pods.dm [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+		else
+			LZ = pick(valid_turfs)
+
+		addtimer(CALLBACK(src, .proc/launch_pod, LZ, crate), (2 SECONDS * num_pods--))
+
+/datum/round_event/resource_pods/proc/launch_pod(turf/LZ, obj/structure/closet/crate/crate)
+	var/obj/structure/closet/crate/spawned_crate = new crate()
+	var/obj/structure/closet/supplypod/pod = new
+	pod.setStyle(pod_style)
+	pod.explosionSize = list(0,0,1,2)
```
Comment (MrMelbert) [inline jollystation_modules/code/modules/events/resource_pods.dm:109]: the pods this event spawns have small explosives, they can be swapped to bluespace pods or have the explosives removed if you don't want rooms getting blasted by a ton of pods

File: jollystation_modules/code/modules/events/resource_pods.dm [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+	var/static/list/allowed_areas
+	if(!allowed_areas)
+		///Places that we shouldn't send crates.
+		var/list/safe_area_types = typecacheof(list(
+			/area/maintenance,
+			/area/tcommsat,
+			/area/ai_monitored,
+			/area/engine,
+			/area/shuttle)
+		)
```
Comment (MrMelbert) [inline jollystation_modules/code/modules/events/resource_pods.dm:134]: ai_monitered includes stuff like the ai sat, armory, and EVA

let me know if more areas should be blacklisted from the steel rain

File: jollystation_modules/code/modules/events/resource_drift.dm (hunk 1 above)
Comment (Jolly-66) [reply jollystation_modules/code/modules/events/resource_drift.dm:66]: if its not as crazy as the 2 major and 1 catastrophic meteor event relly spawned, i think it should be manageable

also having to use your "newly found" resources to fix shit sounds cool

**Pull request merged**